stan/math/rev/functor/apply_scalar_binary.hpp

```diff
@@ -0,0 +1,248 @@
+ * @param f functor to apply to inputs.
+ * @return Eigen object with result of applying functor to inputs.
+ */
+template <typename T1, typename T2, typename F,
+          require_var_matrix_t<T1>* = nullptr,
+          require_std_vector_vt<is_std_vector, T2>* = nullptr,
+          require_std_vector_st<std::is_integral, T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  return f(x, y);
+}
+
+/**
+ * Specialisation for use with a two-dimensional std::vector of integer types
+ * and one `var_value<Matrix>`.
+ *
+ * @tparam T1 Type of first argument to which functor is applied.
+ * @tparam T2 Type of second argument to which functor is applied.
+ * @tparam F Type of functor to apply.
+ * @param x Nested integer std::vector input to which operation is applied.
+ * @param y var value with inner Eigen matrix input to which operation is
+ * applied.
+ * @param f functor to apply to inputs.
+ * @return Eigen object with result of applying functor to inputs.
+ */
+template <typename T1, typename T2, typename F,
+          require_std_vector_vt<is_std_vector, T1>* = nullptr,
+          require_std_vector_st<std::is_integral, T1>* = nullptr,
+          require_var_matrix_t<T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  return f(x, y);
+}
+
+/**
+ * Specialisation for use when the first input is an `var_value<Eigen> type and
+ * the second is a scalar.
+ *
+ * @tparam T1 Type of `var_value<Matrix>` object to which functor is applied.
+ * @tparam T2 Type of scalar to which functor is applied.
+ * @tparam F Type of functor to apply.
+ * @param x Matrix input to which operation is applied.
+ * @param y Scalar input to which operation is applied.
+ * @param f functor to apply to var matrix and scalar inputs.
+ * @return `var_value<Matrix> object with result of applying functor to inputs.
+ *
+ * Note: The return expresssion needs to be evaluated, otherwise the captured
+ *         function and scalar fall out of scope.
+ */
+template <typename T1, typename T2, typename F,
+          require_var_matrix_t<T1>* = nullptr,
+          require_stan_scalar_t<T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  return f(x, y);
+}
+
+/**
+ * Specialisation for use when the first input is an scalar and the second is
+ * an `var_value<Eigen>`.
+ *
+ * @tparam T1 Type of scalar to which functor is applied.
+ * @tparam T2 var value with inner Eigen type to which functor is applied.
+ * @tparam F Type of functor to apply.
+ * @param x Scalar input to which operation is applied.
+ * @param y var matrix input to which operation is applied.
+ * @param f functor to apply to Eigen and scalar inputs.
+ * @return var value with inner Eigen type with result of applying functor to
+ * inputs.
+ *
+ * Note: The return expresssion needs to be evaluated, otherwise the captured
+ *         function and scalar fall out of scope.
+ */
```

Comment:
Can also cut this comment since `unaryExpr` doesn't get called for `varmat` types